PowerEditor/src/ScintillaComponent/Buffer.cpp

```diff
@@ -260,10 +306,43 @@ bool Buffer::checkFileState() // returns true if the status has been changed (it
 	if (_currentStatus == DOC_UNNAMED || isMonitoringOn())
 		return false;
 
+	NppParameters& nppParam = NppParameters::getInstance();
+
 	WIN32_FILE_ATTRIBUTE_DATA attributes{};
 	attributes.dwFileAttributes = INVALID_FILE_ATTRIBUTES;
-	NppParameters& nppParam = NppParameters::getInstance();
-	bool fileExists = doesFileExist(_fullPathName.c_str());
+	bool bWorkerThreadTerminated = true;
+	DWORD dwWin32ApiError = NO_ERROR;
+	BOOL bGetFileAttributesExSucceeded = getFileAttributesExWithTimeout(_fullPathName.c_str(), &attributes, 0, &bWorkerThreadTerminated, &dwWin32ApiError);
+	bool fileExists = (bGetFileAttributesExSucceeded && (attributes.dwFileAttributes != INVALID_FILE_ATTRIBUTES) && !(attributes.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY));
+	if (!fileExists)
+	{
+		if (nppParam.doNppLogNetworkDriveIssue())
+		{
+			wstring issueFn = nppLogNetworkDriveIssue;
+			issueFn += L".log";
+			wstring nppIssueLog = nppParam.getUserPath();
+			pathAppend(nppIssueLog, issueFn);
+			std::string msg = wstring2string(_fullPathName, CP_UTF8);
+			msg += "  in Buffer::checkFileState(), getFileAttributesExWithTimeout returned ";
+			if (bGetFileAttributesExSucceeded)
+				msg += "TRUE";
+			else
+				msg += "FALSE";
+			if (bWorkerThreadTerminated)
+			{
+				msg += ", its worker thread had to be forcefully terminated due to timeout reached!";
+			}
+			else
+			{
+				msg += ", its worker thread finished successfully within the timeout given, ";
+				if (attributes.dwFileAttributes == INVALID_FILE_ATTRIBUTES)
+					msg += "dwFileAttributes == INVALID_FILE_ATTRIBUTES ! (WIN32API Error Code: " + std::to_string(dwWin32ApiError) + ")";
+				else
+					msg += "dwFileAttributes has the FILE_ATTRIBUTE_DIRECTORY flag set!";
+			}
+			writeLog(nppIssueLog.c_str(), msg.c_str());
+		}
+	}

```

Comment:
Use rather:
```cpp
if (!fileExists && nppParam.doNppLogNetworkDriveIssue())
{
    // log here
}
```

Also, this log block should be on line 278/357 (after `#ifndef	_WIN64`), instead being here.